Library/Homebrew/cmd/update

```diff
@@ -0,0 +1,54 @@
+#!/bin/sh # brew-update.sh
+[ -n "$HOMEBREW_VERBOSE" ] && set -x
+
+trap_cleanup()
+{
+  for job in $(jobs -p); do
+    kill -s 2 $job >/dev/null 2>&1
+  done
+
+  for each in $HOMEBREW_PREFIX $(ls -1d $BREW_LIBRARY_DIRECTORY/Taps/*/* 2>/dev/null); do
+    cd "$each"
+    git config --remove-section brew >/dev/null 2>&1
+    git stash pop >/dev/null 2>&1
+  done
+}
+
+trap trap_cleanup HUP INT QUIT ABRT KILL TERM
+
+update_if_necessary()
+(
+  cd "$*"
+  trap trap_cleanup HUP INT QUIT ABRT KILL TERM
+
+  # Build the URI for the API call based on our remote and branch.
+  uri=$(git ls-remote --get-url | sed -e's|git://|https://|' -e's|\.git$||' \
+                                      -e's|github.com|api.github.com/repos|')
+  ref=$(git symbolic-ref HEAD)
+  api_uri=$(echo $uri | sed 's|$|/git|')/$ref
+
+  # We save a few milliseconds with a standard config avoiding the refs API.
+  [ "$ref" = "refs/heads/master" ] && api_uri="$uri/commits/master"
+
+  etag=$(git config brew.etag) # ETag gets cached in .git/config.
```

Comment:
So, I misread your first note there; these get stored each taps' own .git/config already.
And what I meant to say is that this should be something like `git config remote.origin.brew-api-tag` so that it is associated with [ remote "origin" ].